R/plot.R

```diff
@@ -82,3 +82,67 @@ tidy_perf_data <- function(performance_df) {
       )
     )
 }
+
+
+#' Get hyperparameter peformance metrics
+#'
+#' @param trained_model trained model (e.g. from `run_ml()`)
+#'
+#' @return list: dat - dataframe of performance metric for each group of hyperparameters, params - hyperparameters tuned, metric - performance metric used 
+#' @export
+#'
+#' @examples
+#' get_hp_performance(otu_mini_results1$trained_model)
+get_hp_performance <- function(trained_model){
+  params <- trained_model$modelInfo$parameters$parameter
+  metric <- trained_model$metric
+  dat <- trained_model$results %>% dplyr::select(dplyr::all_of(params),dplyr::all_of(metric))
+  params <- params[params %in% names(which(lapply(dat, function(x) length(unique(x))) > 1))]
+  return(list(dat=dat,params=params,metric=metric))
+}
+
+#' Combine hyperparameter performance metrics for several train/test splits
+#'
+#' @param trained_model_lst list of trained models
+#'
+#' @return list: dat - dataframe of performance metric for each group of hyperparameters, params - hyperparameters tuned, metric - performance metric used 
+#' @export
+#'
+#' @examples
+#' combine_hp_performance(list(otu_mini_results1$trained_model,otu_mini_results1$trained_model))
+combine_hp_performance <- function(trained_model_lst){
+  abort_packages_not_installed("purrr")
+  dat_params <- lapply(trained_model_lst, function(x) get_hp_performance(x)) %>% purrr::transpose()
```

Comment:
Can you think of a way to do this without purrr? That way we don't have to add it to Suggests.

Reply:
Not a big deal if not

Reply:
There are some other base r options:
https://stackoverflow.com/questions/15263146/revert-list-structure
e.g.
```
revert_list_str_4 <- function(ls) { # @Josh O'Brien
  # get sub-elements in same order
  x <- lapply(ls, `[`, names(ls[[1]]))
  # stack and reslice
  apply(do.call(rbind, x), 2, as.list) 
}
```

We can change it if you want.

Reply:
oh that's pretty ugly, let's just keep purrr then